openedx/core/djangoapps/user_api/views.py

```diff
@@ -1143,3 +1155,24 @@ class CountryTimeZoneListView(generics.ListAPIView):
     def get_queryset(self):
         country_code = self.request.GET.get('country_code', None)
         return get_country_time_zones(country_code)
+
+
+class HintUserDetailsView(APIView):
+    """
+    DRF APIView to get a username suggestion
+    If the user exists then suggests a new username, otherwise
+    returns the received base username.
+    """
+
+    def get(self, request):
+        """
+        Returns a valid username suggestion checking against the database
+        """
+        data_serializer = HintUsernameSerializer(data=request.query_params)
+        data_serializer.is_valid(raise_exception=True)
+        user_dict = data_serializer.data
+        username = user_dict['username']
+
+        new_username = UsernameGenerator().generate_username(username)
+
+        return Response({'username': new_username}, status=200)
```

Comment:
Nit: unnecessary `status=200`; that's [the default](https://github.com/django/django/blob/stable/1.8.x/django/http/response.py#L102).